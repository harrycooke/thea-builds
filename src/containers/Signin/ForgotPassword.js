import React from 'react';
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { forgotPassword } from "../../actions/authActions";
import classnames from "classnames";
import ArrowIcon from '@material-ui/icons/ArrowBackIos';
import MySnackbarContentWrapper from '../../components/SharedComponents/SnackBar'
import {
  Button,
  CssBaseline,
  TextField,
  Container,
  IconButton,
  Snackbar
} from "@material-ui/core";
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import CircularProgress from '../../components/SharedComponents/CircularProgress';

const styles = theme => ({
  '@global': {
    body: {
      backgroundColor: theme.palette.common.white,
    },
  },
  paper: {
    marginTop: theme.spacing(8),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
  errorText: {
    fontSize: 12,
    fontWeight: "500"
  },
  backButtonDiv: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center'
  },
  backButton: {
    "&:hover": {
      backgroundColor: "transparent"
    },
    fontSize: '16px',
    color: '#3f51b5'
  },
  dialogOverflow: {
    '& .MuiDialog-paper': {
      overflowY: 'visible'
    }
  }
});
class ForgotPassword extends React.Component {

  constructor() {
    super();
    this.state = {
      email: "",
      errors: {},
      open: false,
      setOpen: false,
      snackbarMessage: ''
    };
    this.goBack = this.goBack.bind(this)
    this.handleClose = this.handleClose.bind(this);
  }

  onChange = e => {
    this.setState({ [e.target.id]: e.target.value });
  };

  goBack = () => {
    this.props.history.push('/login')
  }

  onSubmit = e => {
    e.preventDefault();
    const userData = {
      email: this.state.email
    };
    this.props.forgotPassword(userData);
  };

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (nextProps.errors) {
      this.setState({
        errors: nextProps.errors
      });
    }
    if (nextProps.auth.forgotPasswordState && (nextProps.auth.forgotPasswordState !== this.props.auth.forgotPasswordState)) {
      this.handleClick()
      this.setState({ snackbarMessage: nextProps.auth.forgotPasswordState })
      setTimeout(() => {
        this.props.history.push('/login');
      }, 2000)
    }
  }

  handleClick = () => {
    this.setState({ setOpen: true })
  };

  handleClose = (event, reason) => {
    if (reason === 'clickaway') {
      return;
    }

    this.setState({ setOpen: false })
  };
  render() {
    const { classes } = this.props;
    const { errors } = this.state;
    return (
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <div className={classes.paper}>
          <Snackbar
            anchorOrigin={{
              vertical: 'bottom',
              horizontal: 'left',
            }}
            open={this.state.setOpen}
            autoHideDuration={6000}
            onClose={this.handleClose}
          >
            <MySnackbarContentWrapper
              onClose={this.handleClose}
              variant="success"
              message={this.state.snackbarMessage}
            />
          </Snackbar>
          <CircularProgress isLoading={this.props.auth.loading} />
          <div style={{ display: "flex" }}>
            <img src={require("../../assets/images/text-logo.png")} alt="required" style={{width: "100%"}} />
          </div>
          <Typography component="h1" variant="h5">
            Reset Password
          </Typography>
          <form className={classes.form} noValidate onSubmit={this.onSubmit}>
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              id="email"
              label="Email Address"
              name="email"
              autoComplete="off"
              autoFocus
              onChange={this.onChange}
              className={classnames("", {
                invalid: errors.email || errors.emailnotfound
              })}
            />
            <Typography
              color="error"
              className={classes.errorText}
            >
              {errors.email}
              {errors.emailnotfound}
            </Typography>

            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
            >
              Submit
          </Button>
            <div className={classes.backButtonDiv} onClick={this.goBack}>
              <IconButton className={`${classes.button},${classes.backButton}`} aria-label="Go back">
                <ArrowIcon fontSize="small" /> Back
            </IconButton>
            </div>
          </form>
        </div>
      </Container >
    );
  }
}

ForgotPassword.propTypes = {
  auth: PropTypes.object.isRequired,
  errors: PropTypes.any.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth,
  errors: state.errors
});

export default connect(
  mapStateToProps,
  { forgotPassword }
)(withStyles(styles)(ForgotPassword));