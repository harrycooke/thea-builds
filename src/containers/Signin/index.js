import React from 'react';
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { loginUser, setSnackbar } from "../../actions/authActions";
import classnames from "classnames";
import {
  Button,
  CssBaseline,
  TextField,
  FormControlLabel,
  Checkbox,
  Link,
  Grid,
  Container, 
  Snackbar,
} from "@material-ui/core";
import Typography from '@material-ui/core/Typography';
import MySnackbarContentWrapper from '../../components/SharedComponents/SnackBar';
//import { setSnackbar } from "../../actions/caseAction";

import { withStyles } from '@material-ui/core/styles';
import { encodeString, decodeString } from '../../utils/functions';
import Cookies from "universal-cookie";
const cookies = new Cookies();

const styles = theme => ({
  '@global': {
    body: {
      backgroundColor: theme.palette.common.white,
    },
  },
  paper: {
    marginTop: theme.spacing(8),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
  errorText: {
    fontSize: 12,
    fontWeight: "500"
  }
});

export class SignIn extends React.Component {

  constructor() {
    super();
    this.state = {
      email: "",
      password: "",
      errors: {},
      remember: false
    };
    this.forgotPassword = this.forgotPassword.bind(this)
  }

  forgotPassword = () => {
    this.props.history.push("/forgotpassword");
  }

  onChange = e => {
    this.setState({ [e.target.id]: e.target.value });
  };

  onSubmit = e => {
    e.preventDefault();
    const userData = {
      email: this.state.email,
      password: this.state.password
    };
    if(this.state.remember){
      let password = encodeString(this.state.password)
      cookies.set('email', this.state.email, { path: '/' });
      cookies.set('password', password, { path: '/' });
    } else {
      cookies.remove('email', { path: '/' });
      cookies.remove('password', { path: '/' });
    }
    this.props.loginUser(userData);
  };

  handleCheck = () => {
    this.setState({remember: !this.state.remember});
  }

  handleSnackbarClose = () => { 
    this.props.setSnackbar(false);
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (nextProps.auth.isAuthenticated) {
      this.props.history.push("/inbox/cases"); // push user to inbox when they login
    }
    if (nextProps.errors) {
      this.setState({
        errors: nextProps.errors
      });
    }
  }

  componentDidMount() {
    // If logged in and user navigates to Login page, should redirect them to dashboard
    if (this.props.auth.isAuthenticated) {
      this.props.history.push("/inbox/cases");
    }

    let email = cookies.get('email');
    if(email){
      let password = decodeString(cookies.get('password'));
      this.setState({email: email, password: password, remember: true})
    }
  }

  render() {
    const { classes } = this.props;
    const { errors } = this.state;
    return (
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <div className={classes.paper}>

          <div style={{ display: "flex" }}>
            <img src={require("../../assets/images/text-logo.png")} alt="required" style={{width: "100%"}} />
          </div>
          <Typography component="h1" variant="h5">
            Sign In
          </Typography>

          {/* Added the error message for Email and Password fields here */}
          <Typography
              color="error"
              className={classes.errorText}
            >
              {errors.email}
              {errors.emailnotfound}
              {errors.password}
              {errors.passwordincorrect}
            </Typography>

          <form className={classes.form} noValidate onSubmit={this.onSubmit}>
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              id="email"
              label="Email Address"
              name="email"
              autoComplete="off"
              autoFocus
              value={this.state.email}
              onChange={this.onChange}
              className={classnames("", {
                invalid: errors.email || errors.emailnotfound
              })}
            />
            {/* <Typography
              color="error"
              className={classes.errorText}
            >
              {errors.email}
              {errors.emailnotfound}
            </Typography> */}
            <TextField
              variant="outlined"
              margin="normal"
              required
              fullWidth
              name="password"
              label="Password"
              type="password"
              id="password"
              value={this.state.password}
              autoComplete="off"
              onChange={this.onChange}
              className={classnames("", {
                invalid: errors.password || errors.passwordincorrect
              })}
            />
            {/* <Typography
              color="error"
              className={classes.errorText}
            >
              {errors.password}
              {errors.passwordincorrect}
            </Typography> */}
            <FormControlLabel
              control={<Checkbox value="remember" color="primary" />}
              label="Remember me"
              checked={this.state.remember}
              onChange={this.handleCheck}
            />
            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              className={classes.submit}
            >
              Sign In
          </Button>
            <Grid container>
              <Grid item xs style={{ display: "flex", justifyContent: "center" }}>
                <Link style={{ cursor: 'pointer' }} onClick={this.forgotPassword} variant="body2" >
                  Forgot password?
                </Link>
                {/*
                </Grid>
                <Grid item>
                <Link href="/register" variant="body2">
                  {"Don't have an account? Sign Up"}
                </Link>
                */}
              </Grid>
            </Grid>
          </form>
        </div>
        <Snackbar
            anchorOrigin={{
              vertical: 'bottom',
              horizontal: 'left',
            }}
            open={this.props.showSnackbar}
            autoHideDuration={3000}
            onClose={this.handleSnackbarClose}
          >
            <MySnackbarContentWrapper
              onClose={this.handleSnackbarClose}
              variant="success"
              message="Account created successfully."
            />
    </Snackbar>
      </Container>
      
    );
  }
}

SignIn.propTypes = {
  loginUser: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  errors: PropTypes.any.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth,
  errors: state.errors,
  showSnackbar: state.auth.showSnackbar,
});

export default connect(
  mapStateToProps,
  { loginUser, setSnackbar }
)(withStyles(styles)(SignIn));