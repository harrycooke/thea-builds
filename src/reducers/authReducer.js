import {
  SET_CURRENT_USER,
  USER_LOADING,
  USER_DATA,
  CLIENT_DATA,
  PRACTICE_DATA,
  PRACTICE_DATA_REQUEST,
  STRIPE_DATA,
  STRIPE_DATA_REQUEST,
  FORGOT_PASSWORD,
  FORGOT_PASSWORD_REQUEST,
  SPECIALTY_DATA,
  NEW_ACCOUNT,
  NEW_ACCOUNT_REQUEST,
  SHOW_SNACKBAR,
  SHOW_SNACKBAR2,
} from "../actions/types";
const isEmpty = require("is-empty");
const initialState = {
  isAuthenticated: false,
  user: {},
  userdata: {},
  clientdata: {},
  loading: false,
  forgotPasswordState: false,
  specialtyData: [],
  showSnackbar: false,
  showSnackbar2: false,
  practiceData: {},
  stripeData: {}
};
export default function (state = initialState, action) {
  switch (action.type) {
    case SET_CURRENT_USER:
      return {
        ...state,
        isAuthenticated: !isEmpty(action.payload),
        clientdata: {},
        user: action.payload
      };
    case USER_LOADING:
      return {
        ...state,
        loading: true
      };
    case USER_DATA:
      return {
        ...state,
        userdata: action.payload
      }
    case PRACTICE_DATA:
      return {
        ...state,
        practiceData: action.payload,
        loading: false
      }
    case PRACTICE_DATA_REQUEST:
      return {
        ...state,
        loading: action.payload
      };
    case STRIPE_DATA:
      return {
        ...state,
        stripeData: action.payload,
        loading: false
      }
    case STRIPE_DATA_REQUEST:
      return {
        ...state,
        loading: action.payload
      };
    case CLIENT_DATA:
      return {
        ...state,
        clientdata: action.payload
      };
    case FORGOT_PASSWORD:
      return {
        ...state,
        forgotPasswordState: action.payload,
        loading: false
      };
    case FORGOT_PASSWORD_REQUEST:
      return {
        ...state,
        forgotPasswordState: false,
        loading: action.payload
      };
    case SPECIALTY_DATA:
      return {
        ...state,
        specialtyData: action.payload
      };
    case SHOW_SNACKBAR:
      return {
        ...state,
        showSnackbar: action.payload
      }
    case SHOW_SNACKBAR2:
      return {
        ...state,
        showSnackbar2: action.payload
      }
    case NEW_ACCOUNT:
      return {
        ...state,
        newAccountState: action.payload,
        loading: false
      };
    case NEW_ACCOUNT_REQUEST:
      return {
        ...state,
        newAccountState: false,
        loading: action.payload
      };
    default:
      return state;
  }
}