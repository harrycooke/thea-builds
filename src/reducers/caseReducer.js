import {
  CURRENT_CASE,
  SEND_CASE,
  FETCH_ALL_CASES,
  GET_NUM_NEW_MESSAGE,
  CASE_LOADER,
  SHOW_SNACKBAR,
  USER_PATIENTS
} from "../actions/types";
const initialState = {
  sendCases: [],
  currentCase: [],
  readAllCases: [],
  numberOfNewMessage: 0,
  caseLoader: false,
  showSnackbar: false,
  patientList: []
};
export default function (state = initialState, action) {
  switch (action.type) {
    case SEND_CASE:
      return {
        ...state,
        sendCases: action.payload
      };
    case CURRENT_CASE:
      return {
        ...state,
        currentCase: action.payload
      }
    case FETCH_ALL_CASES:
      return {
        ...state,
        readAllCases: action.payload
      }
    case CASE_LOADER:
      return {
        ...state,
        caseLoader: action.payload
      }
    case GET_NUM_NEW_MESSAGE:
      return {
        ...state,
        numberOfNewMessage: action.payload
      }
    case SHOW_SNACKBAR:
      return {
        ...state,
        showSnackbar: action.payload
      }
    case USER_PATIENTS:
      return {
        ...state,
        patientList: action.payload.data
      }
    default:
      return state;
  }
}