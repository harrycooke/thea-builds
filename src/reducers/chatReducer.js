import {
  READ_MESSAGES,
  SEND_MESSAGES
} from "../actions/types";
const initialState = {
  sendMessages: [],
  readMessages: [],
};
export default function(state = initialState, action) {
  switch (action.type) {
    case SEND_MESSAGES:
      return {
        ...state,
        sendMessages: action.payload
      };
    case READ_MESSAGES:
      return {
        ...state,
        readMessages: action.payload
      }
    default:
      return state;
  }
}