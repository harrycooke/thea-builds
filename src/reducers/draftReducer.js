import {
  CURRENT_DRAFT,
  SAVE_DRAFT,
  FETCH_ALL_DRAFT,
  DRAFT_DELETED,
  DRAFT_LOADER
} from "../actions/types";
const initialState = {
  saveDraft: [],
  currentDraft: [],
  readAllDrafts: [],
  isDraftDeleted: false,
  draftLoader: false
};
export default function (state = initialState, action) {
  switch (action.type) {
    case SAVE_DRAFT:
      return {
        ...state,
        saveDraft: action.payload
      };
    case CURRENT_DRAFT:
      return {
        ...state,
        currentDraft: action.payload
      }
    case FETCH_ALL_DRAFT:
      return {
        ...state,
        readAllDrafts: action.payload
      }
    case DRAFT_DELETED:
      return {
        ...state,
        isDraftDeleted: action.payload
      }
    case DRAFT_LOADER:
      return {
        ...state,
        draftLoader: action.payload
      }
    default:
      return state;
  }
}