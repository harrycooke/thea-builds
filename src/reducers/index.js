import { combineReducers } from "redux";
import authReducer from "./authReducer";
import chatReducer from "./chatReducer";
import caseReducer from "./caseReducer";
import draftReducer from "./draftReducer";
import errorReducer from "./errorReducer";
const rootReducer = combineReducers({
  auth: authReducer,
  chat: chatReducer,
  case: caseReducer,
  draft: draftReducer,
  errors: errorReducer
});

export default (state, action) => (
  action.type === 'USER_LOGOUT'
    ? rootReducer(undefined, action)
    : rootReducer(state, action)
)
