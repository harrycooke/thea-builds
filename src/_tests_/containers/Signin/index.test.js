import React from 'react';
import {shallow} from 'enzyme';
import ReactDOM from'react-dom';
import TestRenderer from 'react-test-renderer'; // ES6
import {SignIn, onChange} from '../../../containers/Signin/index';
import reducers from '../../../reducers';
// import {state} from '../state';

// import {commonFormValidation, commonFormOnUpdate} from '../test/shared/shouldBehaveLikeForm';

describe('SignIn ', () => {
// describe('<SignIn />', function() {    
  beforeEach = () => {
    this.setState = ({
        email: "",
        password: "",
        errors: {}
      });
      const email = {...this.email.bind(this)};
      this.state = this.state.bind(this);
    };


it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(SignIn, div);
});

it('renders React components, traversing a ReactElement tree and represents as JSON object', () => {
  const renderer = TestRenderer.create(SignIn);
  renderer.toJSON();
});

describe('SignIn', () => {
  it('should render correctly in "debug" mode', () => {
    const component = (SignIn);
    expect(component).toMatchSnapshot();
  });
});

    // describe('render error messages', () => {
    //     // commonFormValidation.bind(this)(SignIn);
    //     test('render Email error message', () => {
    //         const component = shallow(<SignIn {...this.email} /> );                  //check using the format as password test below
    //         //Define all other fields
    //         component.find('button').simulate('click');
    //         component.update();
    //         expect(component.text()).toEqual(
    //             expect.stringContaining('Email field is required' || 'Email is invalid')
    //         );
    //     });

    //     test('render Password error message', () => {
    //         const component = shallow(SignIn, {...this.state.password} );
    //         //Define all other fields
    //         component.find('button').simulate('click');
    //         component.update();
    //         expect(component.text()).toEqual(
    //             expect.stringContaining('Incorrect username or password.')
    //         );
    //     });
    // });

    // describe('change events update states', () => {
    //     // commonFormOnUpdate.bind(this)(SignIn);
    //     test('update Email state', () => {
    //     const component = shallow(<form {...this.state.email} />);
    //     component
    //         .find('input[name="email"]')
    //         .simulate('change', onChange({target: {name: 'email', value: 'email'}}));
    //     expect(component.state('fields').email).toEqual('email');
    //     });

    //     test('update Password state', () => {
    //     const component = shallow(<form {...this.password} />);
    //     component
    //         .find('input[name="password"]')
    //         .simulate('change', onChange({target: {name: 'password', value: 'password'}}));
    //     expect(component.state('fields').password).toEqual('password');
    //     });

    // });
});

test('Fake test', ()=>{
    expect(true).toBeTruthy();          //here toBeTruthy is a predifined method whichif receives "true", considers to be expected.
});

