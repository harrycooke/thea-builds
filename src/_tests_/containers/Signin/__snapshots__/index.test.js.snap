// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn  SignIn should render correctly in "debug" mode 1`] = `[Function]`;
